Git Data Model Section

martyschaer opened the pull request:
- Add section about internal data model.
- Please feel free to point out any areas that are unclear, so they can be improved.

martyschaer committed:
Add authors declaration

martyschaer committed:
Decrease page margin

martyschaer committed:
Add section about data model

martyschaer committed:
Fix typos

martyschaer commented:
Improvement: describe how commits refer to their parents, and how branches and tags work.

cyrillbolliger commented on handout.md:
```diff
@@ -169,7 +175,127 @@ the main branch.
+The commit message you entered,
+who created the commit,
+who authored the commit
+^[Author and Committer can be different when merges etc. get involved].
```
The dot should probably be inside the brackets...?

cyrillbolliger commented on handout.md:
````diff
@@ -169,7 +175,127 @@ the main branch.
+```
+
+So in order, the tree object contains the following:
+The file mode ^[indicating the file permissions],
````
"The" should be lower case: https://www.grammarly.com/blog/capitalization-after-colons/

cyrillbolliger commented on handout.md:
```diff
@@ -169,7 +175,127 @@ the main branch.
+Git stores it's content in _objects_.
+
+__`blob`__  
+When you `git add` a file a _blob_ object is created and it gets added to the index.
```
Should we introduce the _index_ somewhere?

cyrillbolliger reviewed (changes requested):
- Excellent intro, thanks a lot for writing!
- As the handout should probably meet some scientific criteria, we should not use opinionated words

cyrillbolliger commented on handout.md:
```diff
@@ -3,9 +3,15 @@ title: Maven & Git
 lot: false
 toc: false
+author:
+- Bolliger Cyrill Alexander
```
```suggestion
- Bolliger Cyrill
```

cyrillbolliger commented on handout.md:
```diff
@@ -169,7 +175,127 @@ the main branch.
+
+^[Image Credit: [git-scm.com](https://git-scm.com/book/en/v2/Git-Internals-Git-Objects)]
+
+Cleverly, git only needs to update the (sub-)trees that actually change with
```
Maybe remove _Cleverly_ to be more scientific?

cyrillbolliger commented on handout.md:
```diff
@@ -169,7 +175,127 @@ the main branch.
+\newpage
+
 ## Data Model
+Git's internal data model is extremely simple and elegant,
```
I thinks it's less opinionated if we write "Git's internal data model is _designed to be_ extremely simple and elegant"

martyschaer commented:
>     * Excellent intro, thanks a lot for writing!
> 
>     * As the handout should probably meet some scientific criteria, we should not use opinionated words

I think that's a great idea. I will look into resolving the issues in this either tomorrow or on Wednesday.
(We still have until the 2020-10-22 until the handout needs to be finished).

martyschaer committed:
Resolve some review comments

martyschaer committed:
Add more terms to aspell dictionary

martyschaer committed:
Move XKCD

Move the XKCD about just memorizing a bunch of commands
to the 'Quick Start Guide' section.

martyschaer committed:
Restructure Data Model, introduce trees

cyrillbolliger reviewed (approved):
Thanks a lot!

cyrillbolliger committed:
Introduce the git index

cyrillbolliger committed:
break long lines

martyschaer committed:
Make spellcheck actually work

fix some typos

martyschaer committed:
Move Index section out of data-model into Git Concepts

martyschaer committed:
Add Branches section of data model

martyschaer commented:
@cyrillbolliger I've added a section about branches in the data model section.
With that, I think this branch is ready to merge.

martyschaer commented:
Build Job fails right now when trying to upload the PDF, possibly related to actions/upload-artifact/issues/136
Edit: Worked for now.
